pkg/provider/onepassword/onepassword.go

```diff
@@ -580,6 +587,56 @@ func (provider *ProviderOnePassword) getAllForVault(vaultID string, ref esv1beta
 	return nil
 }
 
+// waitForFunc will wait for OnePassword to _actually_ create/update the secret. OnePassword returns immediately after
+// the initial create/update which makes the next call for the same item create/update a new item with the same name. Hence, we'll
+// wait for the item to exist or be updated on OnePassword's side as well.
+// Ideally we could do bulk operations and handle data with one submit, but that would require re-writing the entire
+// push secret controller. For now, this is sufficient.
+func (provider *ProviderOnePassword) waitForFunc(ctx context.Context, fn func() error) error {
```

Comment:
I'm a bit concerned about waiting for something in providers. Usually, the controller reconciles one resource at a time, so `waitForFunc` blocks the entire reconciliation loop for at most five seconds. The best practice is immediately returning an error and letting the controller requeue the reconciliation request instead of blocking the loop in providers. WDYT? 🤔

Reply:
There is no error though. The problem here is in fact that the provider itself takes longer to make the values actually available making the next reconciliation create a new value instead of updating the existing one.

So I'm not blocking here technically the provider does by taking longer to create the value.

Also 5 seconds is the top that it would block. Hopefully after the first 0.5s wait the value will already been there.

Reply:
I agree with @shuheiktgw, but I see that what you mean, this is 1pass fault... If this was a internally maintained provider I think we should have to follow up with trying to understand the delay on their side and open issues on their repos and/or support tickets. But since this one it community maintained I think it is fine to go with waiting.

Reply:
Alternatively, we can re-write how PushSecret works and instead of pushing every `data` entry separately, we could introduce `bulk` push.

Reply:
`bulk` push could be interesting as a followup or as an option, but overkill for this PR

Reply:
ack